dask/dataframe/io.py

```diff
@@ -750,14 +775,25 @@ def read_hdf(pattern, key, start=0, stop=None, columns=None,
 
     Parameters
     ----------
-    pattern : pattern (string), or buffer to read from. Can contain wildcards
-    key : group identifier in the store. Can contain wildcards
-    start : optional, integer (defaults to 0), row number to start at
-    stop : optional, integer (defaults to None, the last row), row number to
-        stop at
-    columns : optional, a list of columns that if not None, will limit the
-        return columns
-    chunksize : optional, nrows to include in iteration, return an iterator
+    pattern : str
+        String to read from. Can contain wildcards.
+    key : str
+        Group identifier in the store. Can contain wildcards.
+    start : integer, defaults to 0
+        Row number to start at.
+    stop : interger, defaults to None, the last row
+        Row number to stop at.
+    columns : list, optional
+        List of column names to limit the dataframe to.
+    chunksize : int, defaults to 1E6
+        Number of rows to include in iteration, return an iterator.
+    lock : bool or Lock object, default True
+        If True, will apply locks to prevent multiple HDFStore objects from
+        simultaneous reads. If a lock object, will use that to acquire locks.
+        If False, will not use locks.
+    sorted_index_column : bool or str, default False
+        If True, use the sorted index as the key. If True, use the column
```

Comment:
Some users might assume that we're sorting the data for them.  If there is a concise way to avoid this misconception that would be ideal.